src/datatype/index.js

```diff
@@ -121,7 +121,8 @@ export class DataType {
    * @param {string} docId
    */
   async getByDocId(docId) {
-    return deNullify(this.#sql.getByDocId.get({ docId }))
+    const result = this.#sql.getByDocId.get({ docId })
+    return result ? deNullify(result) : result
```

Comment:
This will possibly return `undefined`. I think we want to throw for notFound, and always return a value. Can you open a new issue to address this outside this PR. Ok with this change to keep typescript happy for now, I would have thought this would cause the type tests to fail - we should look at why they aren't

Reply:
oh yeah I'd been meaning to update that CI config